### ISSUE TITLE
Support the ability to set the upstream branch on a push

### ISSUE THREAD

**Issue by kirkilj** [oxygenxml/oxygen-git-plugin#80]
If a local branch doesn't exist yet on the server, the push toolbar arrow hint mentions there's no upstream branch. In such a case I'd like the option to set the upstream to either the same name as the local branch or by entering a new remote/branchName. A dialog box that defaults to the current remote and local branch, but allows other entries to be entered/chosen.

My MVP for this enhancement would be to simply ask if a remote branch should be created with the same name as the local one.

Basically, implement a version of:
`git push --set-upstream remoteName branchName`

**Comment by sorincarbunaru** [comment]
Hi John,

I noticed that JGit (the library upon which we built our add-on) already creates the remote branch, but our UI does not update to show that, because the local repository config file is not updated.

So, if for the initial implementation we just showed a tooltip message "Push to create remote branch X" and after doing that we updated the config file and the tooltips, would that be OK? And perhaps in the future we could show a dialog that lets the user select the remote branch from a list or provide another branch name.
